src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageAccountProvider.cs

```diff
@@ -15,6 +16,7 @@ internal class DefaultStorageAccountProvider : IStorageAccountProvider
         private readonly IStorageCredentialsValidator _storageCredentialsValidator;
         private readonly IStorageAccountParser _storageAccountParser;
         private readonly IServiceProvider _services;
+        private readonly ConcurrentDictionary<string, Task<IStorageAccount>> _accounts = new ConcurrentDictionary<string, Task<IStorageAccount>>();
```

Comment:
The purpose is not to cache values, though that is a side effect - the purpose is to prevent the expensive validate call from occurring more than once. A value cache will not ensure that without extra locking code, which is currently abstracted away behind the ConcurrentDictionary. It's similar to in style to the function loader https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/src/WebJobs.Script/Description/FunctionLoader.cs#L86